k8sclient/k8sclient.go

```diff
@@ -178,18 +176,34 @@ func getCNIConfigFromFile(name string, confdir string) ([]byte, error) {
 	}
 
 	for _, confFile := range files {
-		conf, err := libcni.ConfFromFile(confFile)
-		if err != nil {
-			return nil, fmt.Errorf("Error loading CNI config file %s: %v", confFile, err)
-		}
+		var confList *libcni.NetworkConfigList
+		if strings.HasSuffix(confFile, ".conflist") {
+			confList, err = libcni.ConfListFromFile(confFile)
+			if err != nil {
+				return nil, fmt.Errorf("Error loading CNI conflist file %s: %v", confFile, err)
+			}
+
+			if confList.Name == name {
+				if len(confList.Plugins) == 0 {
+					return nil, fmt.Errorf("CNI config list %s has no networks, skipping", confFile)
```

Comment:
nitpick: s/config list/conflist file
(conflist is used everywhere else)

Reply:
confFile is the same file for `conf` and `conflist` .